### DISCUSSION TITLE
Documentation on using `sample_tree`

### DISCUSSION THREAD

**Discussion by lungsi** [arbor-sim/arbor#1117] [General] [answered]
Hi,

I am trying to figure out on how to use the `sample_tree` function in python. This [documentation](https://arbor.readthedocs.io/en/latest/single_cell.html?) has an example for building a morphology using the `flat_cell_builder`. Could somebody please direct me to either the documentation or example case/s on using the `sample_tree` function?

Thank you,
Lungsi

**Comment by bcumming** [comment, marked as answer]
Hello Lungsi,

The `sample_tree` interface has been replaced with `segment_tree`s, which allow Arbor to model a wider range of morphology characteristics. Unfortunately, as you point out, some of the documentation has not yet been updated. There are some draft documents on the following link. It still requires some work to remove all of the `segment_tree` references.

https://arbor.readthedocs.io/en/pydoc/morphology.html

The `master` branch contains `segment_tree`s, however the most recent release (which you get if installing `pip install arbor` does not.)

You can install a recent version by checking out the GitHub repository and building it yourself, or alternatively via pip:

```bash
pip install git+https://github.com/arbor-sim/arbor.git
```

I hope this helps you get started. Feel free to give us more detail about your morphology-building use case, so that we can give more specific advice, and give us feedback, while I prepare the documentation.
```
